File: Solar Apocalypse (Forge)/src/main/java/com/smushytaco/solar_apocalypse/WorldDayCalculation.java
package com.smushytaco.solar_apocalypse;

import net.minecraft.world.level.Level;

public class WorldDayCalculation {

    public static boolean isOldEnough(Level level, Double days) {
        //if (!SolarApocalypse.INSTANCE.config.getDimensionWhitelist().contains(level.dimension().toString())) return false;
        var worldAgeInMinecraftDays = level.dayTime() / 24000.0;
        return (days < 0.0) ? false : worldAgeInMinecraftDays >= days;
    }

}

File: Solar Apocalypse/src/main/kotlin/com/smushytaco/solar_apocalypse/WorldDayCalculation.kt
package com.smushytaco.solar_apocalypse
import net.minecraft.world.level.Level
object WorldDayCalculation {
    fun Level.isOldEnough(days: Double): Boolean {
        if (!SolarApocalypse.dimensionWhitelist.contains(dimension().identifier().toString())) return false
        val worldAgeInMinecraftDays = this.defaultClockTime / 24000.0
        return days in 0.0..worldAgeInMinecraftDays
    }
}